lib/examples/features.rs

```diff
@@ -1,191 +1,146 @@
 
-    handle.spawn({
-        let future = simple.and_then(|_| markdown).and_then(|_| html);
+    let mut reply = message.text_reply("<b>Bold HTML message</b>");
 
-        future.map_err(|_| ()).map(|_| ())
-    })
+    api.send(reply.parse_mode(ParseMode::Html)).await?;
+    Ok(())
 }
 
-fn test_preview(api: Api, message: Message, handle: &Handle) {
-    let preview = api.send(message.text_reply("Message with preview https://telegram.org"));
+async fn test_preview(api: Api, message: Message) -> Result<(), Error> {
+    api.send(message.text_reply("Message with preview https://telegram.org"))
+        .await?;
 
-    let no_preview = api.send(
-        message
-            .text_reply("Message without preview https://telegram.org")
-            .disable_preview(),
-    );
+    let mut reply = message.text_reply("Message without preview https://telegram.org");
 
-    handle.spawn({
-        let future = preview.and_then(|_| no_preview);
-
-        future.map_err(|_| ()).map(|_| ())
-    })
+    api.send(reply.disable_preview()).await?;
+    Ok(())
 }
 
-fn test_reply(api: Api, message: Message, handle: &Handle) {
-    let msg = api.send(message.text_reply("Reply to message"));
-    let chat = api.send(message.chat.text("Text to message chat"));
-
-    let private = api.send(message.from.text("Private text"));
+async fn test_reply(api: Api, message: Message) -> Result<(), Error> {
+    api.send(message.text_reply("Reply to message")).await?;
+    api.send(message.chat.text("Text to message chat")).await?;
 
-    handle.spawn({
-        let future = msg.and_then(|_| chat).and_then(|_| private);
-
-        future.map_err(|_| ()).map(|_| ())
-    })
+    api.send(message.from.text("Private text")).await?;
+    Ok(())
 }
 
-fn test_forward(api: Api, message: Message, _handle: &Handle) {
-    api.spawn(message.forward(&message.chat));
+async fn test_forward(api: Api, message: Message) -> Result<(), Error> {
+    api.send(message.forward(&message.chat)).await?;
 
-    api.spawn(message.forward(&message.from))
+    api.send(message.forward(&message.from)).await?;
+    Ok(())
 }
 
-fn test_edit_message(api: Api, message: Message, handle: &Handle) {
-    let round_1 = api.send(message.text_reply("Round 1"));
-
-    let duration_1 = Duration::from_secs(2);
-
-    let sleep_1 = Timeout::new(duration_1, handle)
-        .unwrap()
-        .map_err(From::from);
+async fn test_edit_message(api: Api, message: Message) -> Result<(), Error> {
+    api.send(message.text_reply("Round 1")).await?;
 
-    let round_2_api = api.clone();
-    let round_2 = round_1
-        .join(sleep_1)
-        .and_then(move |(message, _)| round_2_api.send(message.edit_text("Round 2")));
+    let when = Instant::now() + Duration::from_secs(2);
+    Delay::new(when).await;
 
-    let duration_2 = Duration::from_secs(4);
-    let sleep_2 = Timeout::new(duration_2, handle)
-        .unwrap()
-        .map_err(From::from);
+    api.send(message.edit_text("Round 2")).await?;
```

Comment:
The behavior here should be:
- Round 1 message is sent
- That same message is later edited to Round 2